ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KVOffset.scala

```diff
@@ -35,13 +36,14 @@ object KVOffset {
     val middle = BigInt(second) << ((end - lowestStart) * 8)
     val lowest = BigInt(third)
     val bytes = (maxValuePlusOne | highest | middle | lowest).toByteArray.drop(1) // this retains leading zeros
-    Offset.fromBytes(bytes)
+    Offset(data.Bytes.fromByteArray(bytes))
   }
 
   def highestIndex(offset: Offset): Long =
-    BigInt(offset.toByteArray.slice(highestStart, middleStart)).toLong
+    BigInt(Offset.unwrap(offset).toByteArray.slice(highestStart, middleStart)).toLong
   def middleIndex(offset: Offset): Long =
-    BigInt(offset.toByteArray.slice(middleStart, lowestStart)).toLong
-  def lowestIndex(offset: Offset): Long = BigInt(offset.toByteArray.slice(lowestStart, end)).toLong
+    BigInt(Offset.unwrap(offset).toByteArray.slice(middleStart, lowestStart)).toLong
+  def lowestIndex(offset: Offset): Long =
+    BigInt(Offset.unwrap(offset).toByteArray.slice(lowestStart, end)).toLong
```

Comment:
We often use two of these together, and the operation is becoming a little costly. Perhaps it'd be useful to have a function that extracts a triple of `(Long, Long, Long)`, and use that above?

Reply:
`middleIndex` and `lowestIndex` are used only once (+1 in the test).
I prefer not touch this, I will let the original designer refactor that.